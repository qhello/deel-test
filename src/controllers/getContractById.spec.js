const request = require("supertest");
const app = require("../app");

describe("getContractById", () => {
  it("should return contract", async () => {
    const res = await request(app).get("/contracts/1").set("profile_id", 5);

    expect(res.statusCode).toEqual(200);
    expect(res.body).toMatchSnapshot();
  });

  it("should return contract", async () => {
    const res = await request(app).get("/contracts/1").set("profile_id", 1);

    expect(res.statusCode).toEqual(200);
    expect(res.body).toMatchSnapshot();
  });

  it("should return 404 if contract doesn't belong to profile", async () => {
    const res = await request(app).get("/contracts/1").set("profile_id", 2);

    expect(res.statusCode).toEqual(404);
  });

  it("should return 404 if contract doesn't exists", async () => {
    const res = await request(app)
      .get("/contracts/foobar")
      .set("profile_id", 5);

    expect(res.statusCode).toEqual(404);
  });
});
