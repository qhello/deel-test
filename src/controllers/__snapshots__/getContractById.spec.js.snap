// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getContractById should return contract 1`] = `
{
  "ClientId": 1,
  "ContractorId": 5,
  "createdAt": "2022-09-27T11:11:32.728Z",
  "id": 1,
  "status": "terminated",
  "terms": "bla bla bla",
  "updatedAt": "2022-09-27T11:11:32.728Z",
}
`;

exports[`getContractById should return contract 2`] = `
{
  "ClientId": 1,
  "ContractorId": 5,
  "createdAt": "2022-09-27T11:11:32.728Z",
  "id": 1,
  "status": "terminated",
  "terms": "bla bla bla",
  "updatedAt": "2022-09-27T11:11:32.728Z",
}
`;
